### PR TITLE
Bluetooth: host: Refactor address and buf to new source files

### PR DESCRIPTION
**Pull request by joerchan**
Refactor bluetooth  address and buf handling out from hci_core and to new source files matching the headers these functions are defined in.

Document the addr.h functions and defines.

### CODE REVIEW

**Review by jhedberg — APPROVED** [review]

**Comment by jhedberg** [comment]
@joerchan there's a compliance issue that needs fixing (redundant blank line at the end of a file)

**Comment by joerchan** [comment]
> @joerchan there's a compliance issue that needs fixing (redundant blank line at the end of a file)

@jhedberg Fixed that.

There was however another issue with ACL_FLOW_CONTROL. I have now moved the declaration of acl_in_pool to buf.c as well.
But that meant I had to expose the host num complete command function and the definition of the acl_data struct.

Also I was a bit puzzled as to how the acl_data type variable worked, so when I figured it out I made another change to it that I think is a bit clearer.

File: include/bluetooth/addr.h [hunk 1]
```diff
@@ -40,44 +41,89 @@ typedef struct {
 	bt_addr_t a;
 } bt_addr_le_t;
 
+/* Bluetooth device "any" address, not a valid address */
```
Comment (Thalley) [inline include/bluetooth/addr.h:44]: Do we want these descriptions to be caught by Doxygen? Shoulnd't they start with `/**` then (Similar to the additional documentation you added in `uuid.h`).

File: include/bluetooth/addr.h [hunk 2]
```diff
@@ -40,44 +41,89 @@ typedef struct {
...
+/* Bluetooth LE device "none" address, not a valid address */
 #define BT_ADDR_LE_NONE ((bt_addr_le_t[]) { { 0, \
 			 { { 0xff, 0xff, 0xff, 0xff, 0xff, 0xff } } } })
 
+/** @brief Compare Bluetooth device addresses.
+ *
+ *  @param a First Bluetooth device address to compare
+ *  @param b Second Bluetooth device address to compare
+ *
+ *  @return negative value if @a a < @a b, 0 if @a a == @a b, else positive
```
Comment (Thalley) [inline include/bluetooth/addr.h:60]: Hah, this line is so hard to follow with the `@a` references :D (Not wrong, at all, but just fun with all the `@a` interleaved)

File: include/bluetooth/addr.h [hunk 3]
```diff
@@ -40,44 +41,89 @@ typedef struct {
...
 #define BT_ADDR_SET_STATIC(a) ((a)->val[5] |= 0xc0)
 
+/** @brief Create a Bluetooth LE random non-resolvable private address. */
 int bt_addr_le_create_nrpa(bt_addr_le_t *addr);
+
+/** @brief Create a Bluetooth LE random static address. */
 int bt_addr_le_create_static(bt_addr_le_t *addr);
 
+/** @brief Check if a Bluetooth LE address is a random private resolvable
+ *         address.
```
Comment (Thalley) [inline include/bluetooth/addr.h:121]: If you wanted to get this on a single line, you could use the RPA abbreviation here, and then expand it on the next line

File: include/bluetooth/addr.h [hunk 4]
```diff
@@ -87,6 +133,15 @@ static inline bool bt_addr_le_is_rpa(const bt_addr_le_t *addr)
 	return BT_ADDR_IS_RPA(&addr->a);
 }
 
+/** @brief Check if a Bluetooth LE address is valid identity address.
+ *
```
Comment (Thalley) [inline include/bluetooth/addr.h:137]: There's some incoherence with
```
/** @brief 
```
and
```
/**
 * @brief
```

in this PR. 

I'm no longer sure what the ideal way is, since `bluetooth.h` was refactored in that regards lately, but some of the other Bluetooth includes weren't...

**Review by Thalley — APPROVED** [review]
A few documentation comments that does not need to be addressed, but probably should be considered.

**Comment by jhedberg** [comment]
@joerchan there are some build tests that seem to be failing..
